lib/pry-doc/pry_ext/show_source_with_c_internals/code_fetcher.rb

```diff
@@ -0,0 +1,125 @@
+require 'fileutils'
+require_relative 'c_file'
+require_relative 'symbol_extractor'
+
+module Pry::CInternals
+  class CodeFetcher
+    include Pry::Helpers::Text
+
+    class << self
+      attr_accessor :ruby_source_folder
+    end
+
+    # normalized
+    def self.ruby_version() RUBY_VERSION.tr(".", "_") end
+
+    self.ruby_source_folder = File.join(File.expand_path("~/.pry.d"), "ruby-#{ruby_version}")
+
+    attr_reader :line_number_style
+    attr_reader :symbol_extractor
+
+    def initialize(line_number_style: nil)
+      @line_number_style = line_number_style
+      @symbol_extractor = SymbolExtractor.new(self.class.ruby_source_folder)
+    end
+
+    def fetch_all_definitions(symbol)
+      infos = self.class.symbol_map[symbol]
+      return unless infos
+
+      "".tap do |result|
+        infos.count.times do |index|
+          result << fetch_first_definition(symbol, index).first << "\n"
+        end
+
+        return [result, full_path_for(infos.first.file)]
+      end
+    end
+
+    def fetch_first_definition(symbol, index=nil)
+      infos = self.class.symbol_map[symbol]
+      return unless infos
+
+      info = infos[index || 0]
```

Comment:
Why not to make `index = 0` on line 39, so it's the default? The check on line 48 can be amended to `if index > 0`, which seems to be quite readable, still.

Reply:
@kyrylo because if i set it to 0 and only display implementations when > 0 then i miss out on `number of implementations: 1` in the output. But maybe that's not a big deal?

Reply:
I don't get it, sorry. If `index = 1`, then it should display `number of implementations: 1` in the output because `if 1 > 0` is true.

Reply:
@kyrylo yes but it won't be showing `number of implementations: 1` if index is 0

Reply:
Makes sense, thanks.

Reply:
Can we just do this:

```rb
def fetch_first_definition(symbol, index=0)
info = infos[index]
result << "#{bold('Number of implementations:')} #{infos.count}\n" if infos.any?
```